CRM/Contribute/Form/AdditionalPayment.php

```diff
@@ -594,52 +501,67 @@ public function processCreditCard($submittedValues) {
         $result = $payment->doPayment($paymentParams);
       }
       catch (\Civi\Payment\Exception\PaymentProcessorException $e) {
-        //set the contribution mode.
-        $urlParams = "action=add&cid={$this->_contactId}&id={$this->_id}&component={$this->_component}";
-        if ($this->_mode) {
-          $urlParams .= "&mode={$this->_mode}";
-        }
-        CRM_Core_Error::displaySessionError($result);
-        CRM_Utils_System::redirect(CRM_Utils_System::url('civicrm/payment/add', $urlParams));
+        Civi::log()->error('Payment processor exception: ' . $e->getMessage());
+        $errorMessage = $e->getMessage();
       }
     }
 
-    if ($result) {
-      $this->_params = array_merge($this->_params, $result);
+    // Payment is not completed
+    if (CRM_Utils_Array::value('payment_status_id', $result) != 1 || $errorMessage) {
```

Comment:
@pradpnayak would it be worth here altering this to use pseduoconstant or similar to get the status_id by label for Completed just in case some random system has completed as id other than 1 perhaps?